tests/services/label.service.spec.ts

```diff
@@ -0,0 +1,138 @@
+import { LabelService } from '../../src/app/core/services/label.service';
+import { Label } from '../../src/app/core/models/label.model';
+import * as LabelConstant from '../constants/label.constants';
+
+let labelService: LabelService;
+let labelList: Label[];
+
+describe('LabelService: parseLabelData()', () => {
+    beforeAll(() => {
+        labelService = new LabelService(null);
+        labelList = labelService.parseLabelData(LabelConstant.LABEL_ARRAY);
+    });
+
+    afterAll(() => {
+        labelService = null;
+    });
+
+    it('should be response.Accepted label', () => {
+        expect(labelList[0].labelCategory).toBe(LabelConstant.RESPONSE);
+        expect(labelList[0].labelValue).toBe(LabelConstant.RESPONSE_ACCEPTED);
+        expect(labelList[0].labelColor).toBe(LabelConstant.COLOR_RESPONSE_ACCEPTED);
+    });
+
+    it('should be severity.Low', () => {
+        expect(labelList[1].labelCategory).toBe(LabelConstant.SEVERITY);
+        expect(labelList[1].labelValue).toBe(LabelConstant.SEVERITY_LOW);
+        expect(labelList[1].labelColor).toBe(LabelConstant.COLOR_SEVERITY_LOW);
+    });
+
+    it('should be type.FunctionalityBug', () => {
+        expect(labelList[2].labelCategory).toBe(LabelConstant.TYPE);
+        expect(labelList[2].labelValue).toBe(LabelConstant.TYPE_FUNCTIONALITY_BUG);
+        expect(labelList[2].labelColor).toBe(LabelConstant.COLOR_TYPE_FUNCTIONALITY_BUG);
+    });
+});
+
+describe('LabelService: isDarkColor()', () => {
+    beforeEach(() => {
+        labelService = new LabelService(null);
+    });
+
+    afterEach(() => {
+        labelService = null;
+    });
+
+    it('should be true for dark color', () => {
+        expect(labelService.isDarkColor(LabelConstant.COLOR_DARK)).toBeTruthy();
+    });
+
+    it('should be false for light color', () => {
+        expect(labelService.isDarkColor(LabelConstant.COLOR_LIGHT)).toBeFalsy();
+    });
+});
+
+describe('LabelService: setLabelStyle()', () => {
+    beforeEach(() => {
+        labelService = new LabelService(null);
+    });
+
+    afterEach(() => {
+        labelService = null;
+    });
+
+    it('should be dark color background with light color text', () => {
+        expect(labelService.setLabelStyle(LabelConstant.COLOR_DARK))
+            .toEqual(LabelConstant.DARK_BG_LIGHT_TEXT);
+    });
+
+    it('should be light color background with dark color text', () => {
+        expect(labelService.setLabelStyle(LabelConstant.COLOR_LIGHT))
+            .toEqual(LabelConstant.LIGHT_BG_DARK_TEXT);
+    });
+});
+
+describe('LabelService: getColorOfLabel()', () => {
+    beforeEach(() => {
+        labelService = new LabelService(null);
+    });
+
+    afterEach(() => {
+        labelService = null;
+    });
+
+    it('should be correct label color for Severity.Low', () => {
+        expect(labelService.getColorOfLabel(LabelConstant.SEVERITY_LOW))
+            .toEqual(LabelConstant.COLOR_SEVERITY_LOW);
```

Comment:
Just a heads up, not necessary need to change it but there are 2 places where you declared this value at. 1 in the service folder and 1 in the testing folder.